src/meepgeom.cpp

```diff
@@ -1757,44 +1777,80 @@ void fragment_stats::update_stats_from_material(material_type mat, size_t pixels
   }
 }
 
+void fragment_stats::compute_overlaps_from_tree(geom_box_tree t, std::vector<double> &overlaps) {
+  if (!t || !geom_boxes_intersect(&t->b, &box))
+    return;
+
+  if (t->nobjects > 0) {
+    geom_box intersection;
+    geom_box_intersection(&intersection, &t->b, &box);
+
+    for (int i = 0; i < t->nobjects; ++i) {
+      if (geom_boxes_intersect(&intersection, &t->objects[i].box)) {
+        geom_box shifted_box = {vector3_minus(intersection.low, t->objects[i].shiftby),
+                                vector3_minus(intersection.high, t->objects[i].shiftby)};
+        double overlap = box_overlap_with_object(shifted_box, *t->objects[i].o, tol, maxeval);
+        if (overlap > 0.0) {
+          for (int j = 0; j < geom.num_items; ++j) {
+            if (t->objects[i].o == geom.items + j) {
+              overlaps[j] += overlap;
```

Comment:
If this loop becomes a limiting factor, we could do the lookup in another way, e.g. a hash table or an array of indices sorted by pointer address.